### ISSUE TITLE
Evaluation during training takes an hour

### ISSUE THREAD

**Issue by superMDguy** [tensorflow/tensor2tensor#596]
I'm running the latest version of tensor2tensor on a custom dataset. My problem definition file is [here](https://github.com/superMDguy/issue-title-generator/blob/master/issues_problem/new_issues_problem.py). Data generation works fine, and the training seems to work fine with the loss decreasing. However, the evaluation steps during the training takes a really long time, about an hour. This really slows down the overall training time. Is this normal, or is there something I'm doing wrong?

### Training Command
```
t2t-trainer \
  --t2t_usr_dir=./issues_problem \
  --data_dir=./data \
  --problems=issue_to_title \
  --model=transformer \
  --hparams_set=transformer_prepend \
  --hparams='batch_size=1024' \
  --output_dir=./train
```

### Software Versions
```
$ pip list | grep tensor
tensor2tensor (1.5.2)
tensorflow-gpu (1.5.0)
tensorflow-tensorboard (1.5.0)
```

### GPU
GTX 1080Ti, with 11gb of memory

**Comment by martinpopel** [comment]
By default `--eval_steps` is set to 10000, thus effectively using the *whole* (dev) test data for each internal evaluation. You can set it to a lower number to make the internal evaluation faster (and less reliable).
(I haven't tried 1.5.2 yet, so I am not sure if there are any other issues with this. I see you use 1% of the original data as (dev) test data, but I don't know how big your data is and whether one hour is reasonable for fast-decoding 1% of the data.)

**Comment by superMDguy** [comment]
Is there any way to have it run 1 epoch through the test set, instead of specifying the number of steps?

**Comment by martinpopel** [comment]
The internal evaluation cannot take more than one pass over the test data. The 10000 steps was meant as a very high constant to make sure that the whole test data is processed. You can check the log to see what is the actual number of steps (mini-batches) in your test data (there should be a end-of-sequence warning after the last step). As an alternative to lowering the `--eval_steps` (or rather in addition to, in your case), you can also run the internal evaluation less often, e.g. by saving the checkpoints just once per hour with `--save_checkpoints_secs=3600`.

**Comment by superMDguy** [comment]
Ok, thanks for the help. Lowering eval steps worked for me, and I'll try increasing `save_checkpoints_secs`.

**Comment by scribu** [comment]
I found the `--eval_throttle_seconds` parameter, which seems better suited.
